### DISCUSSION TITLE
Wrapper Class, Boxing, Unboxing 차이

### DISCUSSION THREAD

**Discussion by seunghye218** [2024-woowacourse-study/level-interview#5] [Java]
.

**Comment by J-I-H-O** [comment]
Wrapper Class, Boxing, Unboxing

**Reply by le2sky** [reply]
땡이가 말해준게 더 좋겠네용

**Comment by kunsanglee** [comment]
기본 타입(primitive type)을 객체로 다루기 위해서 사용하는 클래스들을 래퍼 클래스(wrapper class)라고 합니다. 

컬렉션 프레임워크 같은 경우, 객체만 저장할 수 있습니다. 예를 들어, ArrayList나 HashMap 같은 컬렉션들은 기본 타입의 값을 직접 저장할 수 없고, 대신 이들의 값을 담고 있는 래퍼 클래스의 객체를 저장해야 합니다. 따라서, 기본 타입의 데이터를 컬렉션 프레임워크에 저장하고자 할 때는 해당 기본 타입의 래퍼 클래스로 감싸서(박싱) 넘겨주어야 합니다.

박싱은 기본 타입 데이터를 래퍼 클래스로 감싸 객체로 변환하는 것이고, 언박싱은 래퍼 클래스를 기본 타입 데이터로 변환하는 것 입니다. 박싱과 언박싱을 불필요하게 할 경우 성능에 문제가 생길 수 있기 때문에 주의해야 합니다.

**Comment by seunghye218** [comment]
wrapper class 는 기본 데이터 타입에 해당하는 객체입니다.  
예시로 Double, Character, Boolean 등이 있습니다.  
기본 타입을 메서드 인수로 전달하거나 컬렉션에 저장할 때 사용합니다.

boxing 은 기본 데이터 유형을 해당 래퍼 클래스 인스턴스로 변환하는 프로세스이고 unboxing은 그 반대 과정입니다.  
예를 들어 `int` 값을 `Integer` 객체로 변환하는 것을 boxing이라고 하고 반대로 변환하는 것을 unboxing이라고 합니다. .
wrapper class의 변수에 기본 값을 할당하면 boxing을 수행합니다. 

wrapper class 인스턴스는 null일 수 있지만 기본 데이터 유형은 그럴 수 없습니다.
